includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -210,6 +210,9 @@ private function update_post_meta( &$coupon ) {
 				case 'email_restrictions' :
 					$updated = update_post_meta( $coupon->get_id(), $meta_key, array_filter( array_map( 'sanitize_email', $value ) ) );
 					break;
+				case 'date_expires' :
+					$updated = update_post_meta( $coupon->get_id(), $meta_key, ( $value ? $value->date( 'Y-m-d' ) : '' ) );
```

Comment:
Stored in local time? The date method will format to local - if this was UTC in 2.6.x this needs different handling based on timestamp (can just be regular date function).